zone/zoning.cpp

```diff
@@ -1215,7 +1215,7 @@ void Client::SetPEQZoneFlag(uint32 zone_id) {
 	}
 }
 
-bool Client::CanBeInZone() {
```

Comment:
It's great we're aggregating validation in one function for this, cleans things up.

I'm thinking this should be used in `Handle_OP_ZoneChange` since that is where we send zone cancels if they do not meet requirements.

We already duplicate some of this logic there and it would make sense to have it call it there as well.

I would say that would eliminate the need to have the checks in the commands, but when zone cancel is called - it actually sends the client back to safe coordinates and sometimes drops them from the ceiling which is another issue that we need to trace down and fix.

Regardless, I would add a check of `CanBeInZone` in `Handle_OP_ZoneChange`